core/node/node_framework/src/implementations/layers/state_keeper/mod.rs

```diff
@@ -123,11 +133,32 @@ impl WiringLayer for StateKeeperLayer {
 
 #[derive(Debug)]
 pub struct StateKeeperTask {
-    io: Box<dyn StateKeeperIO>,
-    executor_factory: Box<dyn BatchExecutorFactory<OwnedStorage>>,
-    output_handler: OutputHandler,
-    sealer: Arc<dyn ConditionalSealer>,
-    storage_factory: Arc<dyn ReadStorageFactory>,
+    state_keeper: ZkSyncStateKeeper,
+}
+
+impl StateKeeperTask {
+    pub fn new(
```

Comment:
Nit: This constructor probably doesn't need to be public (so it can be removed altogether in favor of constructing `state_keeper` at the only callsite.

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3193/commits/36ad7c5c2c4247510c52767f41f08ee7591d02cc